pkgs/apps/dirac/default.nix

```diff
@@ -0,0 +1,131 @@
+{ stdenv, lib, fetchurl, gfortran, cmake, makeWrapper
+, which, openssh, blas, lapack, mpi, boost, exatensor, python3
+} :
+
+assert
+  lib.asserts.assertMsg
+  (!blas.isILP64)
+  "A 32 bit integer implementation of BLAS is required.";
+
+stdenv.mkDerivation rec {
+  pname = "dirac";
+  version = "21.0";
+
+  nativeBuildInputs = [
+    which
+    gfortran
+    cmake
+    python3
+    makeWrapper
+    openssh
+  ];
+
+  # For now, DIRAC is built without the PCMSolver optional dependency.
+  buildInputs = [
+    blas
+    lapack
+    boost
+    exatensor
+  ];
+
+  propagatedBuildInputs = [
+    mpi
+  ];
+
+  src = fetchurl {
+    url = "https://zenodo.org/record/4836496/files/DIRAC-${version}-Source.tar.gz";
+    sha256 = "0vqmin24xqkcp3rvml0dimc4kjmb3m37c3l6bkjy4y6d3xyb41yg";
+  };
+
+  prePatch = ''
+    ls -lah
+  '';
+
+  patches = [
+    ./0001-exatensor-cmake.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace ./cmake/custom/exatensor.cmake \
+    --subst-var-by exatensor ${exatensor}
+
+    patchShebangs .
```

Comment:
That will go through all script and sources. Can we narrow that down too specific scripts or subdirs?

Reply:
Those Python scripts are unfortunately scattered throughout the project (it's a real mess), so, no, I don't think so. :)

Reply:
If you just want to patch python sources, you could use find.
```
find -type f -name "*.py" -exec patchShebangs {} \;
```

Reply:
That is a good solution. Please implement that.

Reply:
However, this will not work, as a significant portion of the Python files do not have an ending (or at least not .py). 🙈  Furthermore, there is a number of .sh files that are also fixed in the process.

Reply:
OK, that is annoying. We could improve the find approach a little bit then:
```
# Shell scripts by suffix
find -type f -name "*.sh" -exec patchShebangs {} \;

# Python scripts by shebangs
find -type f -exec grep '#!/usr/bin/env python' {} \; -exec patchShebangs {} \;
```

Reply:
`find -exec patchShebangs` actually does not work (`find: 'patchShebangs': No such file or directory`, I guess because it is defined as function).

The find version with `grep` does not make it faster. I guess in this case we can keep `patchShebangs .` and rely on it to do its job.

Reply:
IMO, looking at the alternatives, `patchShebangs .` seems the most readable solution for this situation. :)